src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -269,6 +372,15 @@ impl<'tcx> LoanPath<'tcx> {
         LoanPath { kind: kind, ty: ty }
     }
 
+    pub fn ref_node_id<'a>(&self) -> ast::NodeId {
```

Comment:
This needs a comment too -- I guess it returns the "node-id of the base variable or upvar of the path"? I feel like there probably exists a helper like this...but maybe not.